src/EnergyPlus/ScheduleManager.cc

```diff
@@ -2586,11 +2571,7 @@ namespace Sched {
 
     std::vector<Real64> const &ScheduleConstant::getDayVals(EnergyPlusData &state, [[maybe_unused]] int jDay, [[maybe_unused]] int dayofWeek)
     {
-        auto const &s_glob = state.dataGlobal;
-        if ((int)tsVals.size() != Constant::iHoursInDay * s_glob->TimeStepsInHour) {
-            this->tsVals.resize(Constant::iHoursInDay * s_glob->TimeStepsInHour);
-            std::fill(this->tsVals.begin(), this->tsVals.end(), this->currentVal);
-        }
+        assert((int)tsVals.size() == Constant::iHoursInDay * state.dataGlobal->TimeStepsInHour);
```

Comment:
Because I make sure all of the tsVals for ScheduleConstant are set ONCE correctly, this can just be an assert now, which will be removed in release mode